crates/biome_js_analyze/src/semantic_analyzers/nursery/no_invalid_use_before_declaration.rs

```diff
@@ -0,0 +1,215 @@
+use crate::{control_flow::AnyJsControlFlowRoot, semantic_services::SemanticServices};
+use biome_analyze::{context::RuleContext, declare_rule, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_js_syntax::{
+    binding_ext::{AnyJsBindingDeclaration, AnyJsIdentifierBinding},
+    AnyJsExportNamedSpecifier,
+};
+use biome_rowan::{AstNode, SyntaxNodeOptionExt, TextRange};
+
+declare_rule! {
+    /// Disallow the use of variables and function parameters before their declaration
+    ///
+    /// JavaScript doesn't allow use of block-scoped bindings (`let`, `const`, and function parameters) before their declaration.
```

Comment:
```suggestion
    /// JavaScript doesn't allow the use of block-scoped bindings (`let`, `const`, and function parameters) before their declaration.
```
can we use a different word instead of “binding”? It’s not a beginner friendly word